operator/v1/types_console.go

```diff
@@ -106,6 +106,38 @@ type ConsoleCustomization struct {
 	// SVG format preferred
 	// +optional
 	CustomLogoFile configv1.ConfigMapFileReference `json:"customLogoFile,omitempty"`
+	// developerCatalog allows to configure the shown developer catalog categories.
+	// +optional
+	DeveloperCatalog DeveloperConsoleCatalogCustomization `json:"developerCatalog,omitempty"`
+}
+
+// DeveloperConsoleCatalogCustomization allow the cluster admin to configure developer catalog.
+type DeveloperConsoleCatalogCustomization struct {
+	// categories which are shown the in the devconsole.
+	// +optional
+	Categories []DeveloperConsoleCatalogCategory `json:"categories,omitempty"`
+}
+
+// DeveloperConsoleCatalogCategoryMeta defines a category incl. label and filter tags
+// for the developer console catalog.
+type DeveloperConsoleCatalogCategoryMeta struct {
+	// ID is an identifier used in the URL to enable deep linking in console.
+	ID string `json:"id"`
+	// label defines a category display label.
```

Comment:
I believe you also need comments like the following or the user can use value `""` even if the property is required.

```
	// +kubebuilder:validation:Required
	// +kubebuilder:validation:MinLength=1
	// +required
```

@jhadvig might know better :)